web/web.go

```diff
@@ -205,34 +205,34 @@ func New(o *Options) *Handler {
 	router.Get("/targets", readyf(instrf("targets", h.targets)))
 	router.Get("/version", readyf(instrf("version", h.version)))
 
-	router.Get("/heap", readyf(instrf("heap", dumpHeap)))
+	router.Get("/heap", instrf("heap", dumpHeap))
 
 	router.Get("/metrics", prometheus.Handler().ServeHTTP)
 
 	router.Get("/federate", readyf(instrh("federate", httputil.CompressionHandler{
 		Handler: http.HandlerFunc(h.federation),
 	})))
 
-	router.Get("/consoles/*filepath", readyf(instrf("consoles", h.consoles)))
+	router.Get("/consoles/*filepath", instrf("consoles", h.consoles))
```

Comment:
Why is it safe to allow any kind of console page while the server isn't ready yet?

Reply:
I thought it was using some JS to hit the query endpoint. If not, that should indeed be reverted.

Reply:
No, it has built-in query functionality. This should only be available once the TSDB is.

Reply:
Okay, reverted.